src/tools/monaco/validate.ts

```diff
@@ -26,15 +26,17 @@ export function validateDSL(monaco: typeof MonacoEditor, dsl: string): Marker[]
       }
 
       const extraInformation: Marker["extraInformation"] = {};
-      const errorMetadata = singleErr.metadata as any;
-      if (errorMetadata.errorType) {
-        extraInformation.error = errorMetadata.errorType;
-      }
-      ["typeName", "relation"].forEach((field) => {
-        if (errorMetadata[field]) {
-          (extraInformation as any)[field] = errorMetadata[field];
+      const errorMetadata = singleErr.metadata;
```

Comment:
Given that the error arose because we cast as any here I've tried to do away with it, so we check for falsy which narrows down to a known type and then check if the fields exist.

We still have to cast as `any` on 36 because the error types don't actually expose that they have those properties